### PR TITLE
[QoL] Add fusion options to overrides

### PR DESCRIPTION
**Pull request by EmberCM**
## What are the changes?
Create override options for starter fusions and enemy fusions

## Why am I doing these changes?
Make testing easier

## What did change?
Added options in the `overrides.ts` for fusions
```ts
export const STARTER_FUSION_OVERRIDE: boolean = false;
export const STARTER_FUSION_SPECIES_OVERRIDE: Species | integer = 0;
export const OPP_FUSION_OVERRIDE: boolean = false;
export const OPP_FUSION_SPECIES_OVERRIDE: Species | integer = 0;
```

### Screenshots/Videos
STARTER_FUSION_OVERRIDE
starter-fusion
STARTER_FUSION_SPECIES_OVERRIDE
starter-fusion-species
OPP_FUSION_OVERRIDE
opp-fusion
OPP_FUSION_SPECIES_OVERRIDE
opp-fusion-species

## How to test the changes?
Change the new options in `overrides.ts`

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

tests

### CODE REVIEW

**Comment by flx-sta** [comment]
Could you also prepare the method in `overridesHelper.ts` please?

**Review by MokaStitcher — APPROVED** [review]
lgtm

<details>
<summary>made a quick test to make sure the test helper functions work, for reference</summary>

```
game.override
      .enemyFusionSpecies(Species.ABOMASNOW)
      .enableEnemyFusion()
      .enableStarterFusion()
      .starterFusionSpecies(Species.GREAT_TUSK);

    await game.classicMode.startBattle();
    const partyMember = game.scene.getPlayerPokemon()!;
    expect(partyMember.isFusion());
    expect(partyMember.fusionSpecies?.speciesId).toBe(Species.GREAT_TUSK)
    const enemyPokemon = game.scene.getEnemyPokemon()!;
    expect(enemyPokemon.isFusion());
    expect(enemyPokemon.fusionSpecies?.speciesId).toBe(Species.ABOMASNOW)
```

</details>

File: src/test/utils/helpers/overridesHelper.ts [hunk 1]
```diff
@@ -85,6 +85,27 @@ export class OverridesHelper extends GameManagerHelper {
     return this;
   }
 
+  /**
+   * Override the player (pokemon) to be a random fusion
+   * @returns this
```
Comment (DayKev) [inline src/test/utils/helpers/overridesHelper.ts:90]:
```suggestion
   * Override the player pokemon to be a random fusion
   * @returns `this`
```

File: src/test/utils/helpers/overridesHelper.ts [hunk 2]
```diff
@@ -85,6 +85,27 @@ export class OverridesHelper extends GameManagerHelper {
...
+   */
+  enableStarterFusion(): this {
+    vi.spyOn(Overrides, "STARTER_FUSION_OVERRIDE", "get").mockReturnValue(true);
+    this.log("Player Pokemon is a random fusion!");
+    return this;
+  }
+
+  /**
+   * Override the player (pokemon) fusion species
```
Comment (DayKev) [inline src/test/utils/helpers/overridesHelper.ts:99]:
```suggestion
   * Override the player pokemon fusion species
```

File: src/test/utils/helpers/overridesHelper.ts [hunk 3]
```diff
@@ -85,6 +85,27 @@ export class OverridesHelper extends GameManagerHelper {
...
+   * @param species the fusion species to set
+   * @returns this
```
Comment (DayKev) [inline src/test/utils/helpers/overridesHelper.ts:101]:
```suggestion
   * @returns `this`
```

File: src/test/utils/helpers/overridesHelper.ts [hunk 4]
```diff
@@ -232,6 +253,27 @@ export class OverridesHelper extends GameManagerHelper {
     return this;
   }
 
+  /**
+   * Override the enemy (pokemon) to be a random fusion
+   * @returns this
```
Comment (DayKev) [inline src/test/utils/helpers/overridesHelper.ts:258]:
```suggestion
   * Override the enemy pokemon to be a random fusion
   * @returns `this`
```

File: src/test/utils/helpers/overridesHelper.ts [hunk 5]
```diff
@@ -232,6 +253,27 @@ export class OverridesHelper extends GameManagerHelper {
...
+   */
+  enableEnemyFusion(): this {
+    vi.spyOn(Overrides, "OPP_FUSION_OVERRIDE", "get").mockReturnValue(true);
+    this.log("Enemy Pokemon is a random fusion!");
+    return this;
+  }
+
+  /**
+   * Override the enemy (pokemon) fusion species
```
Comment (DayKev) [inline src/test/utils/helpers/overridesHelper.ts:267]:
```suggestion
   * Override the enemy pokemon fusion species
```

File: src/test/utils/helpers/overridesHelper.ts [hunk 6]
```diff
@@ -232,6 +253,27 @@ export class OverridesHelper extends GameManagerHelper {
...
+   * @param species the fusion species to set
+   * @returns this
```
Comment (DayKev) [inline src/test/utils/helpers/overridesHelper.ts:269]:
```suggestion
   * @returns `this`
```

**Review by DayKev — APPROVED** [review]

**Comment by DayKev** [comment]
You can ignore the changes I suggested if no other changes are needed for the PR, otherwise add them in at that point.

**Review by flx-sta — APPROVED** [review]

**Pull request merged**